src/main/java/pl/tomaszdziurko/jvm_bloggers/mailing/BlogSummaryMailGenerator.java

```diff
@@ -70,8 +69,16 @@ public String prepareMailContent(int numberOfDaysBackInThePast) {
         String templateContent =  mailingTemplate.getValue();
         StringTemplate template = new StringTemplate(templateContent);
         template.setAttribute("days", numberOfDaysBackInThePast);
-        template.setAttribute("newPosts", newPostsFromPersonalBlogs.stream().map(BlogPostForMailItem::new).collect(Collectors.toList()));
-        template.setAttribute("newPostsFromCompanies", newPostsfromCompanies.stream().map(BlogPostForMailItem::new).collect(Collectors.toList()));
+        template.setAttribute("newPosts", newPostsFromPersonalBlogs.stream().map(
+            blogPost -> {
+                return BlogPostForMailItem.builder().from(blogPost).withDefaultUTMParameters().build();
```

Comment:
You can replace it with lambda expression

    blogPost -> BlogPostForMailItem.builder().from(blogPost).withDefaultUTMParameters().build()